numpy/lib/nanfunctions.py

```diff
@@ -106,6 +106,45 @@ def _copyto(a, val, mask):
     return a
 
 
+def _remove_nan(arr1d, overwrite_input=False):
+    """
+    Equivalent to arr1d[~arr1d.isnan()], but in a different order
+
+    Presumably faster as it incurs fewer copies
+
+    Parameters
+    ----------
+    arr1d : ndarray
+        Array to remove nans from
+    overwrite_input : bool
```

Comment:
How does that square with `arr1d[~arr1d.isnan()]`, which is likely to have different size"